.github/workflows/citgm-package.yml

```diff
@@ -0,0 +1,229 @@
+          KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+          KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 0
+        ports:
+          - '9092:9092'
+        options: >-
+          --link zookeeper
+      postgres:
+        image: ${{ (inputs.postgres == 'true' || inputs.postgres == true) && 'postgres:11-alpine' || '' }}
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_DB: postgres
+          POSTGRES_PASSWORD: postgres
+        ports:
+          - '5432:5432'
+        options: >-
+          --health-cmd pg_isready --health-interval 10s --health-timeout 5s
+          --health-retries 5
+      mongo:
+        image: ${{ (inputs.mongodb == 'true' || inputs.mongodb == true) && 'mongo:5' || '' }}
+        ports:
+          - '27017:27017'
+      mysql:
+        image: ${{ (inputs.mysql == 'true' || inputs.mysql == true) && 'mysql:8.0' || '' }}
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: 'yes'
+          MYSQL_DATABASE: mysql
+        ports:
+          - '3306:3306'
+        options: >-
+          --health-cmd="mysqladmin ping" --health-interval=10s
+          --health-timeout=5s --health-retries=3
+      redis:
+        image: ${{ (inputs.redis == 'true' || inputs.redis == true) && 'redis:7' || '' }}
+        ports:
+          - '6379:6379'
+        options: '--entrypoint redis-server'
+      
+    steps:
+        - name: Check out Fastify
+          uses: actions/checkout@v4
+          with:
+            persist-credentials: false
+
+        - name: Use Node.js
+          uses: actions/setup-node@v4
+          with:
+            node-version: ${{ inputs.node-version }}
+            cache: 'npm'
+            cache-dependency-path: package.json
+
+        - name: Install Dependencies for Fastify
+          run: |
+            npm install --ignore-scripts
+        - name: Npm Link Fastify
+          run: |
+            npm link
+        - name: Determine repository URL of ${{inputs.package}}
+          uses: actions/github-script@v7
+          id: repository-url
+          with:
+            result-encoding: string
+            script: |
+              const response = await fetch('https://registry.npmjs.org/${{inputs.package}}')
+              const data = await response.json()
+              const repositoryUrl = data.repository.url
+              const result = repositoryUrl.match( /.*\/([a-zA-Z0-9-_]+\/[a-zA-Z0-9-_]+)\.git/)[1]
+              return result
+        - name: Check out ${{inputs.package}}
+          uses: actions/checkout@v4
+          with:
+            repository: ${{ steps.repository-url.outputs.result }}
+            path: dependency
+            persist-credentials: false
+        - name: Install Dependencies for ${{inputs.package}}
+          working-directory: dependency
+          run: |
+            npm install
+        - name: Sym Link Fastify
+          working-directory: dependency
+          run: |
+            npm link fastify
+        - name: Postgres Setup
```

Comment:
I think this should be handled at downstream repo. Maybe having a docker container in the repo executing the commands. Tho this might even be just a pre-test setup in `fastify-postgres`